### PR TITLE
fix datepicker errorMessage, was not showing custom errorMessage when ever input has value

### PR DESCRIPTION
**Pull request by Sotruong-copart**

### CODE REVIEW

File: packages/core/src/components/DatePicker/DatePicker.js [hunk 1]
```diff
@@ -109,7 +109,9 @@ class DatePicker extends Component<DatePickerProps, DatePickerState> {
     }
   }
 
-  getErrorMessage = (value: any, errorMessage: string, showCustomError: boolean, customError: string) => showCustomError && isEmpty(value) ? customError : errorMessage
+  getErrorMessage = (value: any, errorMessage: string, showCustomError: boolean, customError: string) => {
```
Comment (colshacol) [inline packages/core/src/components/DatePicker/DatePicker.js]: `value` is now unused.

File: packages/core/src/components/DatePicker/DatePicker.js [hunk 2]
```diff
@@ -109,7 +109,9 @@ class DatePicker extends Component<DatePickerProps, DatePickerState> {
...
+    return showCustomError && customError ? customError : errorMessage
```
Comment (colshacol) [inline packages/core/src/components/DatePicker/DatePicker.js]: `showCustomError && customError || errorMessage` should suffice. :)

**Comment by colshacol** [comment]
Can we get steps to reproduce the issue that this aims to solve? Or steps to test that this code works and what to expect, as opposed to what you would have received with previous behavior?

**Comment by Sotruong-copart** [comment]
> Can we get steps to reproduce the issue that this aims to solve? Or steps to test that this code works and what to expect, as opposed to what you would have received with previous behavior?

https://g-ops-qa4.copart.com/backoffice/finance-reports#PaymentsReport

Step 1. On "To Date" field, type in date 01/01/2001

Expected Behavior: error Message should show up because "To Date" is an earlier date than "From Date"
Observed Behavior: No error message display

Step 2. Delete 01/01/2001 from 'To Date' field

Expected Behavior, error message should not popup unless it is a required error message. This field is not required, so no error message should show up. 
Observed Behavior: Error Message stating "To Date" should be an earlier date than "From Date"

**Comment by sidharthmehra** [comment]
@Sotruong-copart i believe this issue got closed even before i had a chance to see what's going on.
What version of core-components are you on?
We fixed this in 0.5.20 i believe, the errors were showing up on update late which describes the behavior you mentioned in the comment above.

Also the error messages do not have to be hard warnings, it's up to your application if you need these to be hard warnings or not.

Let me know if you need more info on this.

**Comment by sidharthmehra** [comment]
Here is the PR with this fix: #152

**Comment by Sotruong-copart** [comment]
> @Sotruong-copart i believe this issue got closed even before i had a chance to see what's going on.
> What version of core-components are you on?
> We fixed this bug in version 0.5.20, the errors were showing up one update late which describes the behavior you mentioned in the comment above.
> 
> Also the error messages do not have to be hard warnings, it's up to your application if you need these to be hard warnings or not.
> 
> Let me know if you need more info on this.

Currently using     "@copart/core-components": "^0.34.3",
I will update it to the latest version and check to see if that resolve the problems. Thanks for pointing that out.

**Comment by Sotruong-copart** [comment]
The latest version did not resolve the bug.

**Comment by Sotruong-copart** [comment]
> Is this something that still needs work?

Yes, this still needs to be worked on. I think the pull request will resolve the bug. The datepicker is not showing the custom Error Message when datepicker has value.

**Comment by colshacol** [comment]
Here you can see what is going wrong.

https://zeit.co/2RwPQVxHw2jucf6w0YQAOpvj/g2-ops-component-library/hzhkyknurx?redirect=1

**Pull request closed without merging**